app/models/dialog/container_template_service_dialog.rb

```diff
@@ -0,0 +1,47 @@
+class Dialog
+  class ContainerTemplateServiceDialog
+    def self.create_dialog(label, params)
+      new.create_dialog(label, params)
+    end
+
+    # This dialog is to be used by a container template service
+    def create_dialog(label, params)
+      Dialog.new(:label => label, :buttons => "submit,cancel").tap do |dialog|
+        tab = dialog.dialog_tabs.build(:display => "edit", :label => "Basic Information", :position => 0)
+        add_parameters_group(tab, 0, params)
+        dialog.save!
+      end
+    end
+
+    private
+
+    def add_parameters_group(tab, position, params)
+      tab.dialog_groups.build(
+        :display  => "edit",
+        :label    => "Parameters",
+        :position => position
+      ).tap do |dialog_group|
+        params.each_with_index do |param, index|
+          add_parameter_field(param, dialog_group, index)
+        end
+      end
+    end
+
+    def add_parameter_field(param, group, position)
+      group.dialog_fields.build(
```

Comment:
I do not have access to any parameter data at the moment, but I recall that there was enough information to include regex validation.  Is that still possible?